main.go

```diff
@@ -6,60 +6,126 @@ import (
 	"flag"
 	"fmt"
 	"io/fs"
+	"log"
 	"net"
 	"os"
 	"path/filepath"
 	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	ItemTypeTextFile  = 0
+	ItemTypeDirectory = 1
+	// TODO: add more
 )
 
 func main() {
+	if err := do(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func do() error {
 	ex, err := os.Executable()
 	if err != nil {
-		panic(err)
+		return errors.Wrap(err, "getting executable")
 	}
 	exPath := filepath.Dir(ex)
 	fmt.Println(exPath)
 
 	hostFlagPtr := flag.String("host", "localhost", "host to serve from, defaults to localhost")
 	portFlagPtr := flag.Int("port", 1234, "port to run the gopher server on, defaults to 1234")
-	serveDirFlagPtr := flag.String("directory", exPath+"/sample", "path of directory to serve, defaults to sample folder in CWD")
+	serveDirFlagPtr := flag.String(
```

Comment:
forgot, but yeah i'd add a todo to use https://github.com/spf13/viper or something similar mainly to just explore 3rd party lib usage a bit :)